content/300-guides/300-upgrade-guides/800-upgrade-from-prisma-1/08-upgrade-from-mongodb-beta.mdx

```diff
@@ -0,0 +1,231 @@
+---
+
+## Introduction
+
+This guide aims to help you migrate from the MongoDB Beta on Prisma 1 to MongoDB on Prisma 2. To learn more about the differences between Prisma 1 and Prisma 2, you can refer to [this document](https://www.prisma.io/docs/guides/upgrade-guides/upgrade-from-prisma-1/how-to-upgrade#main-differences-between-prisma-1-and-prisma-version-2x-and-later).
+
+The scope of this guide is to give you the workflow necessary to perform the migration and highlight some of the gotchas you may encounter.
+
+We unfortunately can't cover all possible scenarios or changes required, but this guide should help you on your journey. Join [our community Slack](https://slack.prisma.io/) or create an issue [on Github](https://github.com/prisma/prisma1/issues/new/choose) with any questions.
+
+Please perform this migration on your staging environment before trying this in production!
+
+## Requirements
+
+- Must be running MongoDB 4.2+ as a replica set (MongoDB Atlas does this for you automatically)
+- Node.js 12.6.x or 14.x or 16.x
+- TypeScript 4.1.x
+
+## Installing Prisma 2
+
+In your project directory run the following commands:
+
+```bash
+$ npm install prisma
+$ npx prisma init --datasource-provider=mongodb
+```
+
+This should create the following files:
+
+- `prisma/schema.prisma`: An initial Prisma 2 Schema
+- `.env`: Environment file where you'll store your connection string
+
+<Admonition type="warning">
+
+If you see the following error:
+
+```
+ERROR  File schema.prisma already exists in your project.
+Please try again in a project that is not yet using Prisma.
+```
+
+You have likely a `prisma/` directory in your project already. Rename that directory to something like `_prisma/` and try again
+
+</Admonition>
+
+## Find the Connection String to your MongoDB Database
+
+Next you'll want to find the connection string to your MongoDB database. You should be able to find it in your docker-compose file or on Atlas. It's what you'd pass to MongoDB Compass. The connection string should look something like this:
+
+```bash
+mongodb://<user>:<pass>@<host>:27017
+```
+
+The database that stores application data in Prisma 1 is called `default_default`, so we'll add that to the end of the connection string and update the `DATABASE_URL` key in the `.env` file
+
+```bash
+DATABASE_URL="mongodb://prisma:prisma@localhost:27017/default_default"
+```
+
+## Introspect your MongoDB Database
+
+You're now ready to pull the structure of your database down into your Prisma Schema
+
+```bash
+$ npx prisma db pull
+```
+
+And you should see your Prisma Schema in `prisma/schema.prisma` populated with your models.
+
+<Admonition type="warning">
+
+If you see the following error: `Error in connector: SCRAM failure: Authentication failed.`, try adding `?authSource=admin` to the end of your connection string and trying again.
+
+</Admonition>
+
+If you run into any other errors with introspection, please refer to this documentation.
```

Comment:
By "this document" do you mean further down the page, or is it another doc and you forgot to add a link? :-)

Reply:
Good catch! This should link to @keerlu's guide.

Reply:
Actually, I realized there are no docs on how to handle introspection errors, so I removed this.